url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/model_library/ConsumptionModelsLibrary.java

```diff
@@ -0,0 +1,341 @@
+/*   
+   Copyright 2011-2013 The Cassandra Consortium (cassandra-fp7.eu)
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package eu.cassandra.sim.model_library;
+
+import java.util.ArrayList;
+
+import eu.cassandra.sim.entities.appliances.ConsumptionModel;
+import eu.cassandra.sim.entities.appliances.Tripplet;
+
+/**
+ * Class with various ready-to-use consumption models.
+ * 
+ *  @author Fani A. Tzima (fani [dot] tzima [at] iti [dot] gr)
+ * 
+ */
+public class ConsumptionModelsLibrary {
+	
+	public static ConsumptionModel getConsumptionModelForWashingMachine(String type)
+	{
+		if (!type.equals("p") && !type.equals("q"))
+		{
+			System.err.println("non-existent consumption model");
+			System.exit(15);
+		}
+//		int outerN = 0;
+//		int numberOfPatters = 1;
+//		int[] n =new int[numberOfPatters];
+//		n[0] = 1;
+//		ArrayList[] patterns = new ArrayList[numberOfPatters];
+//		// create patterns
+//		Tripplet t = new Tripplet();
+//		if (type.equals("p"))
+//			t.setV(107.74000000000001);
+//		else if (type.equals("q"))
+//			t.setV(107.74000000000001);
+//		t.setD(10);
+//		t.setS(0);
+//		ArrayList tripplets1 = new ArrayList();
+//		tripplets1.add(t);
+//		patterns[0] = tripplets1;
+//		
+//		return new ConsumptionModel(outerN, n, patterns);
+		
+		if (type.equals("p"))
+		{
+			String message = "{\"n\":0,\"params\":[{\"n\":1,\"values\":[{\"p\":107.74000000000001,\"d\":10,\"s\":0}]}]}";
+			try {
+				return new ConsumptionModel(message, "p");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}	
+		else if (type.equals("q"))
+		{
+			String message = "{\"n\":0,\"params\":[{\"n\":1,\"values\":[{\"q\":107.74000000000001,\"d\":10,\"s\":0}]}]}";
+			try {
+				return new ConsumptionModel(message, "q");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		
+		return null;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 33:76, 81:124, 129:172, 177:220, 225:268.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.